### PR TITLE
added code for daily updates monitoring

### PR DESCRIPTION
**Pull request by RomanMatviiv**

### CODE REVIEW

File: daily-updates-monitoring-infrustructure/check_update_status.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+def lambda_handler(event, context):
+    bkg_text_id = event['bkg_text_id']
+    bkg_updates = event['bkg_updates']
+    expected_time = bkg_updates['expected_time']
+    timeout_minutes = bkg_updates['timeout_minutes']
+    bucket_name = bkg_updates['bucket_name']
+
+    session = boto3.Session(region_name='us-east-1',)
+    table_name = 'monitoring_db_table'
```
Comment (TarasKuzyo) [inline daily-updates-monitoring-infrustructure/check_update_status.py:82]: This line is not used

File: daily-updates-monitoring-infrustructure/check_update_status.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    dynamodb = session.resource('dynamodb')
+    sns = session.client("sns")
+    dynamodb_table = dynamodb.Table(os.getenv('TABLE_NAME'))
+
+    # Get current time in EDT timezone
+    edt_timezone = dateutil.tz.gettz('America/New_York')
+    current_time_edt = datetime.now(tz=edt_timezone)
+    print(current_time_edt)
```
Comment (TarasKuzyo) [inline daily-updates-monitoring-infrustructure/check_update_status.py:90]: Remove debug line

File: daily-updates-monitoring-infrustructure/check_update_status.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+    # Generate alert/failure messages
+    failure_msg = failure_message.replace(
+        '__bkg_text_id__', bkg_text_id
+    ).replace(
+        '__trade_date__', (current_time_edt - timedelta(days=bkg_updates['days_offset'])).date().strftime('%Y%m%d')
```
Comment (TarasKuzyo) [inline daily-updates-monitoring-infrustructure/check_update_status.py:96]: Assign `(current_time_edt - timedelta(days=bkg_updates['days_offset'])).date().strftime('%Y%m%d')` to a separate var and then reuse

File: daily-updates-monitoring-infrustructure/check_update_status.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        )
+        # Get record from dynamo db
+        db_record = get_dynamo_db_record(
+            dynamodb_table,
+            bkg_text_id,
+            (current_time_edt - timedelta(days=bkg_updates['days_offset'])).date().strftime('%Y%m%d')
+        )
+        event_log = json.loads(db_record.get('events_log', "[]"))
+        update_times = [record['modified'] for record in event_log]
+        update_times.append(db_record['modified'])
```
Comment (TarasKuzyo) [inline daily-updates-monitoring-infrustructure/check_update_status.py:119]: Why do you append this record?

**Review by TarasKuzyo — CHANGES_REQUESTED** [review]

File: daily-updates-monitoring-infrustructure/initialize_crontab.py [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+) -> Dict[str, Any]:
+    """
+    Sends Get request to the Metadata API.
+
+    :param url: The endpoint URL
+    :param params: Optional argument to add parameters to the endpoint URL
+    return: Returns a dictionary if the response status code is 200 otherwise raise an Exception.
+    """
+    api_login_name = os.getenv('API_LOGIN_NAME')  # 'Maintainer'
+    api_login_secret = os.getenv('API_LOGIN_SECRET')  # 'CyDMEmLotWsykvOo'
```
[the remark this answers is not in the document]
Comment (RomanMatviiv) [reply daily-updates-monitoring-infrustructure/initialize_crontab.py]: Fixed in the last commit

File: daily-updates-monitoring-infrustructure/check_update_status.py (hunk 1 above)
Comment (RomanMatviiv) [reply daily-updates-monitoring-infrustructure/check_update_status.py]: Fixed in the last commit

File: daily-updates-monitoring-infrustructure/check_update_status.py (hunk 2 above)
Comment (RomanMatviiv) [reply daily-updates-monitoring-infrustructure/check_update_status.py]: Fixed in the last commit

File: daily-updates-monitoring-infrustructure/check_update_status.py (hunk 3 above)
Comment (RomanMatviiv) [reply daily-updates-monitoring-infrustructure/check_update_status.py]: Fixed in the last commit

File: daily-updates-monitoring-infrustructure/check_update_status.py (hunk 4 above)
Comment (RomanMatviiv) [reply daily-updates-monitoring-infrustructure/check_update_status.py:120]: The db_record['events_log'] value includes the whole history of updates for a specific dataset and trading day except the last update time which is stored in the db_record['modified'].
So, we need all these timestamps for checking updates for datasets with intraday update logic.

**Review by TarasKuzyo — APPROVED** [review]

**Pull request merged**